src/main/scala/Database/UserDatabase.scala

```diff
@@ -1,78 +1,129 @@
+          displayUsers(resultSet, List.empty).headOption
+        }
+        else throw new Exception("Unable to connect to database")
+      } match {
+        case Failure(exception) => println(exception.getMessage)
+          None
+        case Success(value) => value
+      }
     }
   }
 
-  def getAll: Future[ListBuffer[User]] = {
+  def getAll: Future[List[User]] = {
     Future {
-      Thread.sleep(100)
-      userList
+      val statement = connection.get.createStatement()
+      val resultSet = statement.executeQuery("SELECT * FROM USER")
+      displayUsers(resultSet, List.empty)
     }
   }
 
   def updateById(id: UUID, newName: String): Future[String] = {
-    Thread.sleep(100)
-
-    def extractingUser(user: Option[User]): String = {
-      val userIndex = userList.indexWhere(_.id == id)
-      user match {
-        case Some(user) => userList.update(userIndex, user.copy(name = newName))
-          s"Username updated at the id $id"
-
-        case None => "Id not found."
+    Future {
+      Try {
+        if (connection.isDefined) {
+          val statement = connection.get.createStatement()
+          val rows = statement.executeUpdate(
+            s"""UPDATE USER
+               | set Name = '$newName'
+               | where Id = '${id.toString}'""".stripMargin)
+          rows
+        }
+        else throw new Exception("Unable to connect to database")
+      } match {
+        case Failure(exception) => s"${exception.getMessage}"
+        case Success(value) if value > 0 => s"Updated entry at id ->$id in Database."
       }
     }
-
-    getById(id).transform {
-      case Success(value) => Success(extractingUser(value))
-      case Failure(exception) => Failure(exception)
-    }
   }
 
   def deleteById(id: UUID): Future[String] = {
     Future {
-      Thread.sleep(100)
-      val userIndex = userList.indexWhere(_.id == id)
-
-      if (userIndex != -1) {
-        userList.remove(userIndex)
-        s"User with ID $id deleted from DB."
-      }
-      else {
-        s"No user found with ID $id in DB."
+      Try {
+        if (connection.isDefined) {
+          val statement = connection.get.createStatement()
+          statement.executeUpdate(s"DELETE FROM USER where Id = '${id.toString}'")
+        }
+        else throw new Exception("Unable to connect to database")
+      } match {
+        case Failure(exception) => s"${exception.getMessage}"
+        case Success(value) if value > 0 => s"Deleted entry with id ->$id from Database."
       }
     }
   }
 
-  def deleteAll(): Future[ListBuffer[User]] = {
+  def deleteAll(): Future[String] = {
     Future {
-      Thread.sleep(100)
-      userList.clear()
-      userList
+      Try {
+        if (connection.isDefined) {
+          val statement = connection.get.createStatement()
+          val rows = statement.executeUpdate("TRUNCATE TABLE USER")
+          rows
+        }
+        else throw new Exception("Unable to connect to database")
```

Comment:
This way of handling the db connection is duplicated in this entire file. Move this to a method and handle it at a common place.